File: src/main/java/com/api/parkingcontrol/dto/ParkingSpotDTO.java
package com.api.parkingcontrol.dto;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.Size;

import lombok.Data;

@Data
public class ParkingSpotDTO {

    @NotBlank
    private String parkingSpotNumber;

    @NotBlank
    @Size(max = 7)
    private String licensePlateCar;

    @NotBlank
    private String brandCar;

    @NotBlank
    private String modelCar;

    @NotBlank
    private String colorCar;

    @NotBlank
    private String responsibleName;

    @NotBlank
    private String apartment;

    @NotBlank
    private String block;

}

File: src/main/kotlin/com/api/parkingcontrol/dto/ParkingSpotDTO.kt
package com.api.parkingcontrol.dto

import java.time.LocalDateTime
import java.util.*
import javax.validation.constraints.NotBlank
import javax.validation.constraints.Size

class ParkingSpotDTO(
    var id: UUID? = null,
    @NotBlank val parkingSpotNumber: String,
    @NotBlank @Size(max = 7) val licensePlateCar: String,
    @NotBlank val brandCar: String,
    @NotBlank val modelCar: String,
    @NotBlank val colorCar: String,
    @NotBlank var registrationDate: LocalDateTime?,
    @NotBlank val responsibleName: String,
    @NotBlank val apartment: String,
    @NotBlank val block: String
) {
}